File: src/main/resources/scalagen/finagleClient.scala
// ----- finagle client

import com.twitter.finagle.{Service => FinagleService}
import com.twitter.finagle.stats.{NullStatsReceiver, StatsReceiver}
import com.twitter.finagle.thrift.ThriftClientRequest
import com.twitter.scrooge.FinagleThriftClient

class FinagledClient(
  {{override}}val service: FinagleService[ThriftClientRequest, Array[Byte]],
  {{override}}val protocolFactory: TProtocolFactory = new TBinaryProtocol.Factory,
  override val serviceName: Option[String] = None,
  stats: StatsReceiver = NullStatsReceiver
) extends {{extends}} with FutureIface {
{{#functions}}
{{function}}
{{/function}}
}

File: src/main/resources/javagen/finagleClient.java
public static class FinagledClient extends {{parent}} implements FutureIface {
  private FinagleService<ThriftClientRequest, Array<Byte>> service;
  private TProtocolFactory protocolFactory /* new TBinaryProtocol.Factory */;
  private StatsReceiver stats;

  public FinagledClient(
      FinagleService<ThriftClientRequest, Array<Byte>> service,
      TProtocolFactory protocolFactory /* new TBinaryProtocol.Factory */,
      StatsReceiver stats) {
  {{#hasParent}}
    super(service, protocolFactory, stats);
  {{/hasParent}}
    this.service = service;
    this.protocolFactory = protocolFactory;
    this.stats = stats;
  }

{{#functions}}
  {{>function}}
{{/function}}
}
